### PR TITLE
Add 010 Editor

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** Graeme Sweet via e-Mail
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.sweetscape.ZeroOneZeroEditor

File: com.sweetscape.ZeroOneZeroEditor.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - mkdir -p /app/bin
+      - mv extract.pl /app/bin/extract.pl
+    sources:
+      - type: git
+        url: https://github.com/lod/unpack-install-jammer.git
+        tag: 0.2.0
+        commit: 41faa5a8c7925db4ff7ca665cca478979d177de4
+
+  - name: 010editor
+    only-arches: x86_64
```
Comment (hfiguiere) [inline com.sweetscape.ZeroOneZeroEditor.yml:35]:
```suggestion
    only-arches: [x86_64]
```

also you need to add a flathub.json

https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

File: com.sweetscape.ZeroOneZeroEditor.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - chmod -x /app/lib/010editor/Data/010Repository.zip
+      - mkdir -p /app/bin 
+      - ln -s /app/lib/010editor/010editor /app/bin/010editor
+      - mkdir -p /app/share/icons/hicolor/128x128/apps/
+      - cp /app/lib/010editor/010_icon_128x128.png /app/share/icons/hicolor/128x128/apps/com.sweetscape.ZeroOneZeroEditor.png
+      - install -Dm0644 com.sweetscape.ZeroOneZeroEditor.desktop /app/share/applications/com.sweetscape.ZeroOneZeroEditor.desktop
+      - install -Dm0644 com.sweetscape.ZeroOneZeroEditor.appdata.xml /app/share/appdata/com.sweetscape.ZeroOneZeroEditor.appdata.xml
+    sources:
+      - type: archive
+        url: https://download.sweetscape.com/010EditorLinux64Installer12.0.1.tar.gz
```
Comment (hfiguiere) [inline com.sweetscape.ZeroOneZeroEditor.yml:51]: is redistribution allowed?

File: com.sweetscape.ZeroOneZeroEditor.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        sha256: f0bb4a37f41c75184b30cb76c3810334993eebb6bc3764adc548c61a7cc7c852
+        strip-components: 0
+      - type: file
+        path: com.sweetscape.ZeroOneZeroEditor.desktop
```
Comment (hfiguiere) [inline com.sweetscape.ZeroOneZeroEditor.yml:55]: it doesn't comme with a .desktop?

File: com.sweetscape.ZeroOneZeroEditor.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mailaender) [reply com.sweetscape.ZeroOneZeroEditor.yml:51]: yes, I asked via e-Mail and got explicit permission

File: com.sweetscape.ZeroOneZeroEditor.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mailaender) [reply com.sweetscape.ZeroOneZeroEditor.yml:55]: No, everything is handled by this Install Jammer executable, which behaves like a Windows setup file. It is very alien on Linux. Next version might have tarballs and maybe `.desktop` files, but probably generated by the application on first start, which also wouldn't help much.

**Comment by Mailaender** [comment]
bot, build com.sweetscape.ZeroOneZeroEditor

File: com.sweetscape.ZeroOneZeroEditor.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      - chmod -x /app/lib/010editor/Data/010Repository.zip
+      - mkdir -p /app/bin 
+      - ln -s /app/lib/010editor/010editor /app/bin/010editor
+      - mkdir -p /app/share/icons/hicolor/128x128/apps/
+      - cp /app/lib/010editor/010_icon_128x128.png /app/share/icons/hicolor/128x128/apps/com.sweetscape.ZeroOneZeroEditor.png
+      - install -Dm0644 com.sweetscape.ZeroOneZeroEditor.desktop /app/share/applications/com.sweetscape.ZeroOneZeroEditor.desktop
+      - install -Dm0644 com.sweetscape.ZeroOneZeroEditor.appdata.xml /app/share/appdata/com.sweetscape.ZeroOneZeroEditor.appdata.xml
+    sources:
+      - type: archive
+        url: https://download.sweetscape.com/010EditorLinux64Installer12.0.1.tar.gz
```
Comment (barthalion) [inline com.sweetscape.ZeroOneZeroEditor.yml:52]: Please configure https://github.com/flathub/flatpak-external-data-checker/ to automate updates.

Comment (Mailaender) [reply com.sweetscape.ZeroOneZeroEditor.yml:52]: Done https://github.com/flathub/com.sweetscape.ZeroOneZeroEditor/commit/2be87b63f169f7da39410f740aa911dc6f6485c2

**Pull request closed without merging**